contracts/deposit-contract/sendDepositTx/sendDeposits_test.go

```diff
@@ -106,50 +106,87 @@ func TestEndtoEndDeposits(t *testing.T) {
 	}
 
+	if len(logs) != validatorsWanted  {
+		t.Fatal("No sufficient number of logs")
+	}
 	
+	//validate deposit data 
+	for i, log := range logs {
+		
+		data := depositData[i]
+
+		loggedPubkey, withCreds, _, loggedSig, index, err := contracts.UnpackDepositLogData(log.Data)
+		if err != nil {
+			t.Fatalf("Unable to unpack logs %v", err)
+		}
+
+		if binary.LittleEndian.Uint64(index) != 0 {
+			t.Errorf("Retrieved merkle tree index is incorrect %d", index)
+		}
+
+		if !bytes.Equal(loggedPubkey, data.PublicKey) {
+			t.Errorf("Pubkey is not the same as the data that was put in %v", loggedPubkey)
+		}
+
+		if !bytes.Equal(loggedSig, data.Signature) {
+			t.Errorf("Proof of Possession is not the same as the data that was put in %v", loggedSig)
+		}
+
+		if !bytes.Equal(withCreds, data.WithdrawalCredentials) {
+			t.Errorf("Withdrawal Credentials is not the same as the data that was put in %v", withCreds)
+		}
+	}
+
 	//Match the deposit count, 
 	depositCount, err := testAcc.Contract.DepositContractCaller.DepositCount(bind.NewCallOpts()).Uint64()
 	if err != nil {
 		t.Fatalf("Unable to retrieve deposit count from deposit contract %v", err)
 	}
-	totalNumberOfDeposits := validatorsWanted * numberOfDeposits
-	if depositCount != totalNumberDeposits {
-		t.Errorf("Incorrect depositCount, expected %v, got %v", totalNumberOfDeposits, depositCount)
+	if depositCount != validatorsWanted  {
+		t.Errorf("Incorrect depositCount, expected %v, got %v", validatorsWanted, depositCount)
 	}
 
-	
-
-	//TODO 
-	
-	//2.validate deposit data and ensure it is accurate and the same data you sent over in a transaction,
-	//type DepositContractDepositEvent struct 
+	// validate merkle root of trie in the deposit contract
+	lock.Lock()
+	defer lock.Unlock()
+
+    //make deposits
+	deposits := map([]*ethpb.Deposit, 0, validatorsWanted)
+	for _, data := range depositData {
+		deposit := &ethpb.Deposit{
+			Data: data,
+		}
+        deposits = append(deposits, deposit)
+	}
 
-	//func (s *Service) ProcessDepositLog(ctx context.Context, depositLog gethTypes.Log) error {
-	depositEventIterator, err := testAcc.Contract.DepositContractFilterer.FilterDepositEvent(bind.NewCallOpts())
+	//generate deposit proof
+	_, root:= helpers.GenerateDepositProof(t, deposits[0:validatorsWanted])
+   
+	//verify merkle proof for each deposit
+	receiptRoot := eth1Data.DepositRoot
+	leaf, err := ssz.HashTreeRoot(deposit.Data)
 	if err != nil {
-		t.Fatalf("Unable to retrieve deposit event iterator %v", err)
+		return errors.Wrap(err, "could not tree hash deposit data")
 	}
-    //how rto iterate with Iterator with counter 
-	for EventIterator.Next  {
-		current := depositEventIterator.Event
-		counter +16 
-		data  := depositData[i]
+	if ok := trieutil.VerifyMerkleProof(
+		receiptRoot,
+		leaf[:],
+		int(eth1Data.DepositCount-1),
+		deposit.Proof,
+	); !ok {
+		return fmt.Errorf(
+			"deposit merkle branch of deposit root did not verify for root: %#x",
+			receiptRoot,
+		)
```

Comment:
@nisdas i would like to validate merkle root of trie in the deposit contract . First of all, i generate deposits from depositData , then i produce a root of trie using GenerateDepositProof, and finally i apply trieutil.VerifyMerkleProof . Does it sound good ?